barretenberg/cpp/src/barretenberg/vm2/simulation/events/nullifier_tree_read_event.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "barretenberg/crypto/merkle_tree/indexed_tree/indexed_leaf.hpp"
+#include "barretenberg/vm2/common/field.hpp"
+#include "barretenberg/world_state/types.hpp"
+
+#include <cstdint>
+#include <vector>
+
+namespace bb::avm2::simulation {
+
+using NullifierTreeLeafPreimage = crypto::merkle_tree::IndexedLeaf<crypto::merkle_tree::NullifierLeafValue>;
```

Comment:
hmm maybe I should create some type aliases like these and use them in our db interfaces. will consider.